fr.esrf.tango.generator.xtend/src/fr/esrf/tango/pogo/generator/python/PythonDeviceHL.xtend

```diff
@@ -264,6 +264,11 @@ def enumClasses(PogoDeviceClass cls) '''
         «cls.setEventCriteria»
         «IF cls.description.filestogenerate.toLowerCase.contains("protected regions")»
         «cls.openProtectedAreaHL("init_device")»
+        «IF !cls.attributes.empty»
+        «FOR attr:cls.attributes»
+        self._«attr.pythonAttributeVariableNameHL» = «attr.defaultValueHL»
+        «ENDFOR»
+        «ENDIF»

```

Comment:
What happens inside protected regions like this when you regenerate code?  For example, if I modified the code below to initialise `self._new_variable`, as below.  And then regenerated form pogo, preserving protected regions, what is end result?

```python
    def init_device(self):
        """Initialises the attributes and properties of the TestAttrName."""
        Device.init_device(self)
        # PROTECTED REGION ID(TestAttrName.init_device) ENABLED START #
        self._max_fft_length = 0.0
        self._health_state = HealthState.OK
        self._receptor_id = 0.0
        self._scan_id = 0.0
        self._speed = 0.0
        self._check_spectrum = (0.0,)
        self._string_spectrum_attr = ('',)
        self._check_image_attr = ((0.0,),)
        self._string_image_attr = (('',),)
        self._new_variable = 123
        # PROTECTED REGION END #    //  TestAttrName.init_device
```

Reply:
These changes remain as is. But if you add a new attribute from POGO and regenerate code then that new attribute won't be initialized in init_device method, as protected regions are not regenerated.

Reply:
OK, thanks.  That seems reasonable.